tools/flang2/flang2exe/llutil.cpp

```diff
@@ -2194,6 +2194,96 @@ metadata_args_need_struct(void)
   return ll_feature_metadata_args_struct(&llvm_get_current_module()->ir);
 }
 
+/**
+ * This function returns true for the types supported
+ * in function make_param_op
+ */
+bool
+should_preserve_param(const DTYPE dtype)
+{
+  switch (DTY(dtype)) {
+  // handled cases
+  case TY_ARRAY:
+    {
+      ADSC *ad = AD_DPTR(dtype);
+      SPTR size_sptr = AD_NUMELM(ad);
+      ISZ_T size = ad_val_of(size_sptr);
+      /* dont preserve zero-sized array, which would be optimized out later */
+      if (size == 0)
+        return false;
+      else
+        return true;
+    }
+  case TY_STRUCT:
+  case TY_BLOG:
+  case TY_SLOG:
+  case TY_LOG:
+  case TY_BINT:
+  case TY_SINT:
+  case TY_INT:
+  case TY_REAL:
+  case TY_INT8:
+  case TY_LOG8:
+  case TY_DBLE:
+  case TY_QUAD:
+  case TY_CMPLX:
+  case TY_DCMPLX:
+  case TY_CHAR:
+    return true;
+  // unsupported cases
+  case TY_WORD:
+  case TY_DWORD:
+  case TY_HOLL:
+  case TY_NCHAR:
+    return false;
+  default:
+    assert(0, "should_preserve_param(dtype): unexpected DTYPE", 0, ERR_Fatal);
+    return false;
+  }
+}
+
+OPERAND *
+make_param_op(SPTR sptr)
+{
+  OPERAND *oper;
+  DTYPE dtype = DTYPEG(sptr);
+
+  switch (DTY(dtype)) {
```

Comment:
Isn't this `switch` supposed to handle `TY_ARRAY` and `TY_STRUCT` as well?

Reply:
OK, now I understand why. It would be useful to add a comment here to explain the inconsistency between this function and `should_preserve_param` immediately above.

Reply:
Yes, you are right. I shall add comment here.